### DISCUSSION TITLE
How to have poetry downgrade dependencies when python requirements is too low instead of having to upgrade python version?

### DISCUSSION THREAD

**Discussion by kkawabat** [python-poetry/poetry#7080] [General] [answered]
I am trying to install some data science dependencies for a project of mine with the command `poetry add matplotlib numpy pandas scipy`

I am using python 3.7 so I have this in my `pyproject.toml`

> [tool.poetry.dependencies]
> python = "~3.7"

When I try to add the dependencies I ran into a solverproblemError because the current version of scipy (1.9.3) does not support 3.7

> The current project's Python requirement (3.7) is not compatible with some of the required packages Python requirement:
>     - scipy requires Python >=3.8, so it will not be satisfied for Python 3.7
>   
>   Because no versions of scipy match >1.9.3,<2.0.0
>    and scipy (1.9.3) requires Python >=3.8, scipy is forbidden.
>   So, because md-pipeline-repositories depends on scipy (^1.9.3), version solving failed.

Is there a way to just have poetry search for an older version of scipy and install that instead of just stopping the installation?

I see on this [site](https://docs.scipy.org/doc/scipy/dev/toolchain.html) that they list out what version of scipy is compatible with what version of python 

![image](https://user-images.githubusercontent.com/9159267/203431409-d1f86f04-eda4-47d9-9340-4f6651a6f5fe.png)

I can manually just do `poetry add scipy==1.6` and it seems to work fine. I don't understand why poetry won't just downgrade the version of scipy that's compatible with 3.7.

**Comment by dimbleby** [comment, marked as answer]
> Is there a way to just have poetry search for an older version of scipy and install that?

> I can manually just do poetry add scipy==1.6 and it seems to work fine.

seems you have answered your own question

#707

**Reply by kkawabat** [reply]
Not really, because I have to do the manual work of looking for the spec sheet for all the dependencies to figure out which one is compatible with python.

I thought the whole point of poetry is to automate the dependency solving. If I have to manually figure out what version all the dependencies is necessary for my version of python it defeats the purpose of using `poetry add` no?

**Reply by dimbleby** [reply]
Again, #707.

**Reply by kkawabat** [reply]
Can you explain how that addresses my issue?  I'm not asking WHY this is happening I'm asking for a solution to my problem. Just because I copied and pasted similar messages that poetry threw in 707 it doesn't mean that the threads are duplicated.  That thread isn't even about `poetry add` it's about `poetry init` and sometimes I don't have all the dependencies listed at the creation of a project. 

Again my issue is that I don't want to search through all my dependencies and figure out the exact version of each dependency that would work for my python version. The whole point of poetry is managing dependencies.

**Reply by neersighted** [reply]
This is a duplicate of #707 -- the issue covers both `add` and `init` as it is the same code path. Currently to just do what you want, you basically want `poetry add 'scipy==*'`. Trying to infer a working constraint (instead of assuming `@latest`) is a different ask; that is what #707 tracks.

**Reply by dimbleby** [reply]
#707 is an open bug report exactly describing what you want to do and noting that it doesn't work.

It doesn't "address" your issue - it's an *open* bug.

I wouldn't hold your breath on it being fixed, either.  It's been ~five~ four years now: more or less by definition everyone who has ever encountered it is a python developer, yet no-one has cared enough to do anything about it.

Perhaps you will be the one?
